src/rpc_server.rs

```diff
@@ -1696,6 +1735,14 @@ pub trait RPC {
     /// ```
     async fn restart_miner(token: rpc_auth::Token) -> RpcResult<()>;
 
+    /// Provide a PoW-solution to the current block proposal.
+    ///
+    /// Returns true iff the provided solution satisfies the proof-of-work
+    /// threshold of the next block. False otherwise. False can be returned if
+    /// the solution comes in too late in which case either a new block
+    /// proposal is stored by the client, or no block proposal is known at all.
```

Comment:
I'm confused by this explanation. Maybe something like the following works better? I might also have misinterpreted this function's purpose due to my confusion.

```rust
/// Suggest a solution for the current block proposal's proof-of-work puzzle.
///
/// In case the suggestion is no valid solution, of if there is no current block
/// proposal, or if a new block proposal has been received in the meantime, the
/// suggestion is rejected and `false` is returned.
```

I'm not sure about the “new block proposal has been received in the meantime” part. I understand it neither in the original docstring nor in my suggestion.

Reply:
I don't have all the context that comes into play here. With that caveat out of the way:

- I'm generally in favor of owning up to inherent complexity instead of delegating difficult decisions by default.[^difficult] If we, with years of accumulated knowledge, don't know what's right (or usually right), can we really expect other people to know?
- If there are odd cases or rough edges that are not dealt with, they should at least be outlined as clearly as possible, for example in the documentation.

[^difficult]: _Allowing_ different decisions at the same time is even better.